tests/test_mpi.py

```diff
@@ -945,6 +945,23 @@ def test_avoid_haloudate_if_flowdep_along_other_dim(self):
             assert np.allclose(f.data_ro_domain[0, 5:], [4.8, 4.8, 4.8, 4.8, 3.6], rtol=R)
             assert np.allclose(g.data_ro_domain[0, 5:], [4.8, 4.8, 4.8, 4.8, 2.], rtol=R)
 
+    @pytest.mark.parallel(mode=2)
+    def test_haloudate_scheduling_with_time_plus_no_time_varying_funcs(self):
```

Comment:
nitpicking: would you mind changing the name to `test_unmerge_haloupdate_if_no_locindices` ? I find it a bit clearer (and ideally we'd need it clear since there's no docstring).